addons/mail/static/src/core/common/message_search_hook.js

```diff
@@ -16,9 +17,18 @@ export function searchHighlight(searchTerm, target) {
     }
     const htmlDoc = new DOMParser().parseFromString(target, "text/html");
     for (const term of searchTerm.split(" ")) {
-        const regexp = new RegExp(`(${term})`, "gi");
+        // escaping single quote using xpath concat()
+        const split = term.toLowerCase().split("'");
+        let concat = split.map(s => `'${s}'`).join(', "\'", ');
+        let upperConcat = concat.toUpperCase();
+        if (split.length > 1) {
+            concat = `concat(${concat})`;
+            upperConcat = `concat(${upperConcat})`;
+        }
```

Comment:
Can't you just do `term.replace("'", "\'")`?

Reply:
Have you tried to wrap in double quotes instead of singles in the XPath expression?

Reply:
And escape the `"` inside the term with a simple replace?

Reply:
Same problem, if i use double quotes then I get a traceback when I try searching for a double quote, even if I replace it with `term.replace('"', '\"')`

Reply:
actually should be `term.replace("'", "\\'")` but still does not work